File: tai-stt/app/gatekeeper/gatekeeper_decision.py
from dataclasses import dataclass
from enum import Enum


class RejectionCategory(str, Enum):
    NONE = "NONE"
    NOISE = "NOISE"
    UNINTELLIGIBLE = "UNINTELLIGIBLE"


@dataclass(frozen=True)
class GatekeeperDecision:
    accepted: bool
    reason: str
    suspicion_score: int
    rejection_category: RejectionCategory

    def to_dict(self) -> dict:
        return {
            "accepted": self.accepted,
            "reason": self.reason,
            "suspicionScore": self.suspicion_score,
            "rejectionCategory": self.rejection_category.value,
        }

File: tai-stt-listener/src/main/java/com/toteuch/tai/stt/listener/gatekeeper/GatekeeperDecision.java
// SPDX-License-Identifier: GPL-3.0-only
package com.toteuch.tai.stt.listener.gatekeeper;

public record GatekeeperDecision(
        boolean accepted, String reason, int suspicionScore, RejectionCategory rejectionCategory) {
    public static GatekeeperDecision accepted(int suspicionScore) {
        return new GatekeeperDecision(true, "ACCEPTED", suspicionScore, RejectionCategory.NONE);
    }

    public static GatekeeperDecision noise(String reason) {
        return new GatekeeperDecision(false, reason, 999, RejectionCategory.NOISE);
    }

    public static GatekeeperDecision unintelligible(String reason) {
        return new GatekeeperDecision(false, reason, 999, RejectionCategory.UNINTELLIGIBLE);
    }

    public static GatekeeperDecision suspicious(String reason, int suspicionScore) {
        return new GatekeeperDecision(
                false, reason, suspicionScore, RejectionCategory.UNINTELLIGIBLE);
    }
}
